application/search/filters.py

```
@@ -280,18 +280,44 @@ class QueryFilters:
-        return v
+                    # Check if the geometry passed has GeoJSON format instead
+                    geometry_str = str(geometry).strip()
+                    if geometry_str.startswith('{"type"') or geometry_str.startswith(
+                        '"{"type"'
+                    ):
+                        raise InvalidGeometry(
+                            "Invalid geometry format. Expected WKT format "
+                            "(e.g., 'POLYGON((...))'), but received GeoJSON format. "
+                            "(e.g., '{'type': 'Polygon', 'coordinates': [[...]]}'. "
+                            "Please convert your GeoJSON to WKT format."
+                        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)